client/src/signup/Signup.jsx

```diff
@@ -17,8 +19,23 @@ const Signup = () => {
 
   const handleSubmit = (e) => {
     e.preventDefault();
-    // TODO: implement form submission
-    console.log("Form submitted:", formData);
+    //POSTリクエストを送信する
+    const data = {
+      username: document.querySelector('input[name="username"]').value,
+      weight: document.querySelector('input[name="weight"]').value,
+      email: document.querySelector('input[name="email"]').value,
+      password: document.querySelector('input[name="password"]').value,
```

Comment:
L5 で `formData` を状態として持っているから、以下のコードで実装するべきだと思う
```suggestion
      username: formData.username,
      weight: formData.weight,
      email: formData.email,
      password: formData.password,
```